packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -376,6 +385,7 @@ const DataGrid = injectIntl((props) => {
     const key = event.keyCode;
     switch (key) {
       case KeyCode.KEY_UP:
+        if (!hasVisibleColumnHeaders && nextRow - 1 === 0) break;
```

Comment:
I would think you could move this logic to this block of code
    if (nextCol < 0 || nextRow < 0) {
      event.preventDefault(); // prevent the page from moving with the arrow keys.
      return;
    }

Your current implementation does not account for preventing the default behavior so that the page does not scroll.  The basis of that logic is already lower in the file and checking for the top boundary.  You would just have to modify the condition to also handle the hidden column header use case.

Reply:
Updated here: 766fbf53f940f41b8d382040e42f2f83feccf791